File: it/expected-structures/code/default/csharp/Test/Type.cs
using Newtonsoft.Json;
using System;
using System.Text;

namespace Test {
  [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
  public class Type {
    [JsonConstructor]
    public Type() {
    }

    public override Int32 GetHashCode() {
      Int32 result = 1;
      return result;
    }

    public override Boolean Equals(Object other) {
      Type o = other as Type;

      if (o == null) {
        return false;
      }

      return true;
    }

    public override String ToString() {
      StringBuilder b = new StringBuilder();

      b.Append("Type");
      b.Append("(");
      b.Append(")");

      return b.ToString();
    }
  }
}

File: it/expected-structures/code/default/java/test/Type.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;

public class Type {
  @JsonCreator
  public Type() {
  }

  @Override
  public int hashCode() {
    int result = 1;
    return result;
  }

  @Override
  public boolean equals(final Object other) {
    if (other == null) {
      return false;
    }

    if (!(other instanceof Type)) {
      return false;
    }

    @SuppressWarnings("unchecked")
    final Type o = (Type) other;

    return true;
  }

  @Override
  public String toString() {
    final StringBuilder b = new StringBuilder();

    b.append("Type");
    b.append("(");
    b.append(")");

    return b.toString();
  }

  public void typeMethod() {
  }

  public static class Builder {
    public Type build() {

      return new Type();
    }
  }
}
